### PR TITLE
fix(htl): use new htlengine

### PR DESCRIPTION
**Pull request by tripodsan**
Issue:

## What I did
- use updated htlengine 4.0.0
- replaced `BrowserRuntime.js` with ootb Runtime from htlengine
- use proper `rootContext` and `context` when compiling htl scripts
- revert temporary fix in `list.html` to include correct template

## How to test
- build and click around

### CODE REVIEW

File: htl-loader/src/index.js [hunk 1]
```diff
@@ -16,6 +16,11 @@ const loaderUtils = require('loader-utils');
 const { Compiler } = require('@adobe/htlengine');
 
 async function run(content) {
+}
+
+module.exports = async function compile(content) {
+  // const options = loaderUtils.getOptions(this);
```
Comment (CodeByAlex) [inline htl-loader/src/index.js]: Can this comment be removed?

File: htl-loader/src/index.js [hunk 2]
```diff
@@ -24,21 +29,17 @@ async function run(content) {
...
     .withRuntimeVar('wcmmode')
     .withRuntimeVar('component')
     .withModuleImportGenerator(modGen);
 
   // compile the script to a executable template function
-  const template = await compiler.compileToString(content);
+  const template = await compiler.compileToString(content, this.context);
 
-  console.log(template);
+  // console.log(template);
```
Comment (CodeByAlex) [inline htl-loader/src/index.js]: Can this console log comment be removed?

File: htl-loader/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   ],
   "license": "Apache-2.0",
   "dependencies": {
-    "@adobe/htlengine": "./htl-loader/adobe-htlengine-3.3.1-pre1.tgz",
+    "@adobe/htlengine": "4.0.0",
```
Comment (CodeByAlex) [inline htl-loader/package.json]: Nice!

File: htl-loader/src/index.js (hunk 2 above)
Comment (tripodsan) [reply htl-loader/src/index.js]: ok

File: htl-loader/src/index.js (hunk 1 above)
Comment (tripodsan) [reply htl-loader/src/index.js]: ok

**Pull request merged**